editor/plugins/gdextension/cpp_scons/template/addon/SConstruct_addon

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+
+# This file is for building as a Godot GDExtension.
+ARGUMENTS["precision"] = "double"
+env = SConscript("src/godot-cpp/SConstruct")
+
+# Add source files.
+env.Append(CPPPATH=["./,src/,__UPDIR_DOTS__/"])
+sources = Glob("*.cpp") + Glob("src/*.cpp") + Glob("__UPDIR_DOTS__/*.cpp")
+
+env.Append(CPPDEFINES=["GDEXTENSION"])
+
+bin_path = "bin/"
+extension_name = "__LIBRARY_NAME__"
+debug_or_release = "release" if env["target"] == "template_release" else "debug"
+
+if not "arch_suffix" in env:
+    env["arch_suffix"] = env["arch"]
+
+if env["target"] == "editor":
+    env.Append(CPPPATH=["__UPDIR_DOTS__/editor/"])
+    sources += Glob("__UPDIR_DOTS__/editor/*.cpp")
+
+if env["target"] in ["editor", "template_debug"]:
+    try:
+        doc_data = env.GodotCPPDocData("src/gen/doc_data.gen.cpp", source=Glob("doc_classes/*.xml"))
+        sources.append(doc_data)
+    except AttributeError:
+        print("Not including class reference as we're targeting a pre-4.3 baseline.")
+
+# Create the library target (e.g. lib__LIBRARY_NAME__.linux.debug.x86_64.so).
+if env["platform"] == "macos":
+    library = env.SharedLibrary(
+        "{0}/lib{1}.{2}.{3}.framework/{1}.{2}.{3}".format(
```

Comment:
`.framework` is not required or useful on macOS, but _is_ required on iOS.
But we're currently trying to figure out if `.framework` is even worth generating at all, because if in doubt, the godot exporter upgrades `.dylib` to `.framework` anyway. This should probably be cleared up before merging this PR. I'm waiting for feedback from @bruvzg for this :)

Reply:
@aaronfranke We've finally confirmed that `GDExtension` does not need to use `.framework` (Godot exporter handles upgrade and signing). 
We've simplified the godot-cpp-template accordingly (see https://github.com/godotengine/godot-cpp-template/pull/78). Would probably a nice simplification for your PR too!